setup.cfg

```diff
@@ -23,3 +23,19 @@ include_trailing_comma = True
 spark_options =
     spark.sql.session.timeZone: UTC
     spark.driver.bindAddress: 127.0.0.1
+
+[mypy]
+# suppress errors about unsatisfied imports
+ignore_missing_imports=True
+
+# be strict
+warn_return_any = True
+strict_optional = True
+warn_no_return = True
+warn_redundant_casts = True
+warn_unused_ignores = True
+disallow_any_generics = True
+
+disallow_untyped_defs = True
+check_untyped_defs = True
+disallow_untyped_calls = True
```

Comment:
I had never used Mypy before. Could you explain a little more about these configurations here and the decision to set up them? I mean, just an overview of the benefits.

Reply:
This first configuration will guarantee a better typing for the distribution of the library that will satisfy the users who will use it in a typed context. It will also ensure that we will type as much as possible.

You can find all config settings [here](https://mypy.readthedocs.io/en/stable/config_file.html)

### warn_return_any

It avoids us to return `Any` in a function declared to return a more specific type (e.g. int, str, List[T], CassandraColumn).

Code example:
```python
def foo() -> int:  # mypy will warn here
    return bar()

def bar() -> Any:
    return 10
```

### strict_optional

mypy disallow the `None` compatibility with all of the types

Code example:
```python
def foo() -> int:
    return None  # mypy will warn this

# If we want to return `None` we have to explicit annotate this intention
def bar() -> Optional[int]:
    return None  # mypy won't warn this
```

### warn_no_return

Code example:
```python
def foo(arg: int) -> bool:  # mypy will warn here, this function has two paths but just one path retuns something
    if arg > 10:
        return True
```

### warn_redundant_casts

Code example:
```python
foo: int = 42
foo: int = 10  # mypy will warn here, it has already infered the type
```

### warn_unused_ignores

Code example:
```python
def foo() -> int:
    return 42

x = foo()  # type: ignore  # This is unused ignore, mypy won't warn this line without the comment
```

### disallow_any_generics

Code example:
```python
T = TypeVar('T')

class Foo(Generic[T]):
    ...

foo: Foo = Foo()  # mypy will warn here
foo: Foo[int] = Foo()  # mypy won't warn here
```

### disallow_untyped_defs

Code example:
```python
def foo(arg_one, arg_two):  # mypy will warn here, `foo` does not have type annotations
    ...

def bar(arg_one: str, arg_two):  # mypy will warn here, `bar` does not have type annotations for all of its arguments
    ...
```

### check_untyped_defs

Code example:
```python
def foo():  # mypy will check this function
    return bar('str')  # mypy will warn here because it knows the `bar` function accepts only `int` as a first parameter

def bar(arg_one: int) -> int:
    return arg_one
```

### disallow_untyped_calls

Code example:
```python
def foo(arg: int) -> None:
    bar(arg)  # mypy will warn here because `bar` does not have type annontations

def bar(arg):
    ...
```

### ignore_missing_imports

Code example:
```python
import my_lib.foo  # mypy will warn here because they can't resolve the dependency
# `ignore_missing_imports` disable this behavior
```